From 1b37416d5ec3a5070bcc0616a5fa79a1261c46f4 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Fri, 7 Feb 2014 13:46:59 -0800
Subject: [PATCH 447/974] # User Ronny Meeus <ronny.meeus@alcatel-lucent.com> #
 Date 1386156065 -3600 # Wed Dec 04 12:21:05 2013 +0100 # Node ID
 a0615a4c69c8749f43f285c4fdb756dbf572a38c # Parent
 2ba522e9bb1949c61327c3cca0afedb7c09a3702 alu change: enable unicast filter
 for all relevant network interfaces

[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/net/ethernet/octeon/ethernet.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/ethernet/octeon/ethernet.c b/drivers/net/ethernet/octeon/ethernet.c
index 2411eef..1072b7d 100644
--- a/drivers/net/ethernet/octeon/ethernet.c
+++ b/drivers/net/ethernet/octeon/ethernet.c
@@ -974,6 +974,7 @@ static int cvm_oct_probe(struct platform_device *pdev)
 				priv->tx_lockless = priv->tx_multiple_queues && !disable_lockless_pko;
 				dev->netdev_ops = priv->tx_lockless ?
 					&cvm_oct_sgmii_lockless_netdev_ops : &cvm_oct_sgmii_netdev_ops;
+				dev->priv_flags |= IFF_UNICAST_FLT;
 				priv->gmx_base = CVMX_GMXX_RXX_INT_REG(interface_port, interface);
 				strcpy(dev->name, "xaui%d");
 				break;
@@ -988,6 +989,7 @@ static int cvm_oct_probe(struct platform_device *pdev)
 				priv->tx_lockless = priv->tx_multiple_queues && !disable_lockless_pko;
 				dev->netdev_ops = priv->tx_lockless ?
 					&cvm_oct_sgmii_lockless_netdev_ops : &cvm_oct_sgmii_netdev_ops;
+				dev->priv_flags |= IFF_UNICAST_FLT;
 				priv->gmx_base = CVMX_GMXX_RXX_INT_REG(interface_port, interface);
 				strcpy(dev->name, "eth%d");
 				break;
@@ -996,6 +998,7 @@ static int cvm_oct_probe(struct platform_device *pdev)
 				priv->tx_lockless = priv->tx_multiple_queues && !disable_lockless_pko;
 				dev->netdev_ops = priv->tx_lockless ?
 					&cvm_oct_sgmii_lockless_netdev_ops : &cvm_oct_sgmii_netdev_ops;
+				dev->priv_flags |= IFF_UNICAST_FLT;
 				priv->gmx_base = CVMX_AGL_GMX_RXX_INT_REG(0);
 				strcpy(dev->name, "agl%d");
 				break;
@@ -1004,6 +1007,7 @@ static int cvm_oct_probe(struct platform_device *pdev)
 				priv->tx_lockless = priv->tx_multiple_queues && !disable_lockless_pko;
 				dev->netdev_ops = priv->tx_lockless ?
 					&cvm_oct_spi_lockless_netdev_ops : &cvm_oct_spi_netdev_ops;
+				dev->priv_flags |= IFF_UNICAST_FLT;
 				priv->gmx_base = CVMX_GMXX_RXX_INT_REG(interface_port, interface);
 				strcpy(dev->name, "spi%d");
 				break;
@@ -1013,6 +1017,7 @@ static int cvm_oct_probe(struct platform_device *pdev)
 				priv->tx_lockless = priv->tx_multiple_queues && !disable_lockless_pko;
 				dev->netdev_ops = priv->tx_lockless ?
 					&cvm_oct_rgmii_lockless_netdev_ops : &cvm_oct_rgmii_netdev_ops;
+				dev->priv_flags |= IFF_UNICAST_FLT;
 				priv->gmx_base = CVMX_GMXX_RXX_INT_REG(interface_port, interface);
 				strcpy(dev->name, "eth%d");
 				break;
-- 
2.6.2

